src/Microsoft.AspNet.Diagnostics.Elm/ElmCaptureMiddleware.cs

```diff
@@ -48,11 +45,11 @@ public ElmCaptureMiddleware(RequestDelegate next, ILoggerFactory factory, IOptio
         /// Takes the info from the given HttpContext and copies it to an HttpInfo object
         /// </summary>
         /// <returns>The HttpInfo for the current elm context</returns>
-        private static HttpInfo GetHttpInfo(HttpContext context, Guid requestId)
+        private static HttpInfo GetHttpInfo(HttpContext context)
         {
             return new HttpInfo()
             {
-                RequestID = requestId,
+                RequestID = context.GetFeature<IHttpRequestIdentifierFeature>().TraceIdentifier,
```

Comment:
You can't tell if you're inside a logging scope, correct? I wouldn't conditionally create one. Do or don't.

Conditionally creating a TraceIdentifier would be good, but if you do that you should assign it back to the request.